File: Python/Mean Filter.py
import numpy as np
import matplotlib.pyplot as plt
from scipy.ndimage import convolve
import imageio.v2 as imageio


img = imageio.imread('Enter image path here')
img = img.astype(np.uint8)

filter_size = 6
kernel = np.ones((filter_size, filter_size), dtype=np.float32) / (filter_size ** 2)

filtered_img = np.zeros_like(img)
for c in range(img.shape[2]):
    filtered_img[:, :, c] = convolve(img[:, :, c], kernel, mode='nearest')

plt.figure(figsize=(10, 5))

plt.subplot(1, 2, 1)
plt.imshow(img)
plt.title('Original Image')
plt.axis('off')

plt.subplot(1, 2, 2)
plt.imshow(filtered_img)
plt.title('Filtered Image with Mean Filter')
plt.axis('off')

plt.tight_layout()
plt.show()

File: MATLAB/Mean Filter.m
img = imread('/MATLAB Drive/noisy_image.jpg');
filterSize = 3;
h = fspecial('average', filterSize);
[rows, cols, numChannels] = size(img);
filteredImg = zeros(rows, cols, numChannels, 'like', img);
for channel = 1:numChannels
    filteredImg(:, :, channel) = imfilter(img(:, :, channel), h, 'replicate');
end
figure;
subplot(1, 2, 1);
imshow(img);
title('Original Image');
subplot(1, 2, 2);
imshow(filteredImg);
title('Filtered Image with Mean Filter');
